aries_vcx_core/src/anoncreds/credx_anoncreds.rs

```diff
@@ -211,13 +303,91 @@ impl BaseAnonCreds for IndyCredxAnonCreds {
         sig_type: Option<&str>,
         config_json: &str,
     ) -> VcxCoreResult<(String, String)> {
-        let _ = (issuer_did, schema_json, tag, sig_type, config_json);
-        Err(unimplemented_method_err("credx issuer_create_and_store_credential_def"))
+        let issuer_did = issuer_did.to_owned().into();
+        let schema = serde_json::from_str(schema_json)?;
+        let sig_type = sig_type.map(serde_json::from_str).unwrap_or(Ok(SignatureType::CL))?;
+        let config = serde_json::from_str(config_json)?;
+
+        let schema_seq_no = match &schema {
+            Schema::SchemaV1(s) => s.seq_no,
+        };
+
+        let cred_def_id =
+            credx::issuer::make_credential_definition_id(&issuer_did, schema.id(), schema_seq_no, tag, sig_type)?;
+
+        let options = json!({
+            "retrieve_type": false,
+            "retrieve_value": true,
+            "retrieve_tags": false,
+        })
+        .to_string();
+
+        // If cred def already exists, return it
+        if let Ok(cred_def) = self
+            .wallet
+            .get_wallet_record(CATEGORY_CRED_DEF, &cred_def_id.0, &options)
+            .await
+        {
+            return Ok((cred_def_id.0, cred_def));
+        }
+
+        // Otherwise, create cred def
+        let (cred_def, cred_def_priv, cred_key_correctness_proof) =
+            credx::issuer::create_credential_definition(&issuer_did, &schema, tag, sig_type, config)?;
+
+        let str_cred_def = serde_json::to_string(&cred_def)?;
+
+        // Store stuff in wallet
+        self.wallet
+            .add_wallet_record(CATEGORY_CRED_DEF, &cred_def_id.0, &str_cred_def, None)
```

Comment:
as per note above, we not store the data under new custom "type". Every wallet item is identified by tuple "id + type"  pair. Since we store different data, we want to make sure not to mix things up.

Reply:
I had a type there, now fixed "not -> now"

Anyway, yeah I was just a vocal observation that the wallet interface requires specifying "id" and "type" to store/lookup some record (though vdrtools defines the values of "type" on its own, while here we have to do it ourselves).